build.sbt

```diff
@@ -252,6 +264,96 @@ lazy val connectCommon = (project in file("spark-connect/common"))
       PB.gens.java -> (Compile / sourceManaged).value,
       PB.gens.plugin("grpc-java") -> (Compile / sourceManaged).value
     ),
+  ).configureUnidoc(
+    generatedJavaDoc = false,
+    generateScalaDoc = false
+  )
+
+lazy val connectClient = (project in file("spark-connect/client"))
+  .dependsOn(connectCommon % "compile->compile;test->test;provided->provided")
+  .settings(
+    name := "delta-connect-client",
+    commonSettings,
+    releaseSettings,
+    Compile / compile := runTaskOnlyOnSparkMaster(
+      task = Compile / compile,
+      taskName = "compile",
+      projectName = "delta-connect-client",
+      emptyValue = Analysis.empty.asInstanceOf[CompileAnalysis]
+    ).value,
+    Test / test := runTaskOnlyOnSparkMaster(
+      task = Test / test,
+      taskName = "test",
+      projectName = "delta-connect-client",
+      emptyValue = ()
+    ).value,
+    publish := runTaskOnlyOnSparkMaster(
+      task = publish,
+      taskName = "publish",
+      projectName = "delta-connect-client",
+      emptyValue = ()
+    ).value,
+    crossSparkSettings(),
```

Comment:
Do we need the `crossSparkSettings` since we are only compiling against spark master?

Reply:
@andreaschat-db I think we still need it to get the

`    scalaVersion := scala213,
    crossScalaVersions := Seq(scala213),
    targetJvm := "17",
    resolvers ++= Seq(
      "Spark master staging" at "https://repository.apache.org/content/groups/snapshots/"
    ),
    Compile / unmanagedSourceDirectories += (Compile / baseDirectory).value / "src" / "main" / "scala-spark-master",
    Test / unmanagedSourceDirectories += (Test / baseDirectory).value / "src" / "test" / "scala-spark-master",
    Antlr4 / antlr4Version := "4.13.1",
    java17TestSettings,`
    
    in the [crossSparksSettings()](https://github.com/delta-io/delta/blob/master/build.sbt#L151)

Reply:
yes, we still need it because it sets the `scalaVersion`,  `antlr4Version`, source directories for the shimming, etc